packages/evm/src/eof/container.ts

```diff
@@ -0,0 +1,229 @@
+import { KIND_CODE, KIND_DATA, KIND_TYPE, MAGIC, TERMINATOR, VERSION } from './constants'
+import { EOFError, validationError } from './errors'
+
+class StreamReader {
```

Comment:
Just for my understanding: what is the main purpose respectively the EOF specific part of this extra class? Mainly to throw the EOF related errors? Can this class get a short 1-2 line code doc summary describing the purpose? *If* the class is EOF specific I would also tend to give this an expression in the name and do `EOFStreamReader`, `StreamReader` sounds as "if it could do everything" [TM]. 😋

Reply:
Maybe one can also give this validation some part in the name to get this more expressive? `EOFStreamValidationReader`? Open for better names! 🙂